src/Ledger/Chain/Properties/GovDepsMatch.lagda

```diff
@@ -0,0 +1,104 @@
+\begin{code}[hide]
+{-# OPTIONS --safe #-}
+
+open import Ledger.Transaction
+open import Ledger.Abstract
+
+module Ledger.Chain.Properties.GovDepsMatch
+  (txs : _) (open TransactionStructure txs)
+  (abs : AbstractFunctions txs) (open AbstractFunctions abs)
+  where
+\end{code}
+% If the module name changes, change the following macro to match!
+\newcommand{\ChainPropGov}{Chain/Properties/GovDepsMatch}
+
+\begin{code}[hide]
+open import Ledger.Certs govStructure
+open import Ledger.Chain txs abs
+open import Ledger.Enact govStructure
+open import Ledger.Epoch txs abs
+open import Ledger.Epoch.Properties.GovDepsMatch txs abs
+open import Ledger.Ledger txs abs
+open import Ledger.Ledger.Properties txs abs
+open import Ledger.Ledger.Properties.GovDepsMatch txs abs
+open import Ledger.Prelude hiding (map) renaming (mapˢ to map)
+
+module _
+  {b   : Block }
+  {nes : NewEpochState}
+  {cs  : ChainState}
+  where
+  open Block b; open ChainState cs
+  open NewEpochState
+  open EPOCH-Body (EpochStateOf cs) renaming (epsLState to csLState)
+  open EnactState ens using (pparams)
+  pp = pparams .proj₁
+  open PParams pp using (maxRefScriptSizePerBlock)
+\end{code}
+
+\begin{theorem}[%
+  \LedgerMod{\ChainPropGov.lagda}{\AgdaModule{\ChainPropGov{}}}:
+  \AgdaFunction{govDepsMatch} is invariant of \CHAIN{} rule%
+  ]\
+  \label{thm:ChainGovDepsMatch}
+
+  \begin{itemize}
+    \item \textit{Informally}.
+      Fix a \Block{} \ab{b}, a \ChainState{} \ab{cs}, and a \NewEpochState{} \ab{nes}.
+      Let \AgdaFunction{csLState} be the ledger state of \ab{cs}.
+      Recall, a \ChainState{} has just one field, \AgdaField{newEpochState}~:~\NewEpochState{}.
+      Consider the chain state \AgdaFunction{cs'} defined as follows:
+\begin{code}
+  cs' : ChainState
+  cs' .newEpochState =
+    record { lastEpoch   = nes .lastEpoch
+           ; epochState  = record (EpochStateOf cs) {ls = LStateOf nes}
+           ; ru          = nes .ru }
+\end{code}
+      That is \AgdaFunction{cs'} is essentially \ab{nes}, but the \EpochState{} field is
+      set to the \AgdaField{epochState} of \ab{cs} with the exception of the
+      \LState{} field, which is set to that of \ab{nes}.
+      \\[4pt]
+      Let \ab{utxoSt} and \ab{utxoSt'} be the respective \UTxOState{}s of the ledger
+      states of \ab{cs} and \AgdaFunction{cs'}, respectively, and let \ab{govSt} and \ab{govSt'}
+      be their respective \GovState{}s.
+      \\[4pt]
+      Assume the following conditions hold:
+      \begin{itemize}
+      \item the hypothesis about \AgdaFunction{removed'} needed for and described in \cref{lem:EpochGovDepsMatch},\\[4pt]
```

Comment:
I'm not a fan that this is a forward reference, instead of there being a backwards one (since this is the theorem maybe it'd make more sense to explain it here)